mypyc/genops.py

```diff
@@ -457,6 +487,19 @@ def gen_return(self, builder: 'IRBuilder', value: Value) -> None:
         self.outer.gen_return(builder, value)
 
 
+class GeneratorNonlocalControl(BaseNonlocalControl):
+    def gen_return(self, builder: 'IRBuilder', value: Value) -> None:
+        # Assign an invalid next label number so that the next time __next__ is called, we jump to
+        # the case in which StopIteration is raised.
+        builder.assign_to_target(builder.fn_info.generator_class.next_label_target,
+                                 builder.add(LoadInt(-1)),
+                                 builder.fn_info.fitem.line)
+        # Raise a StopIteration containing a field for the value that should be returned.
+        builder.add(RaiseStandardError(RaiseStandardError.STOP_ITERATION, value,
+                                       builder.fn_info.fitem.line))
+        builder.add(Unreachable())
```

Comment:
There is one subtle snag here:
The definition of returns from a generator say that return should raise a `StopIteration` but the `StopIteration` shouldn't be caught by `except` blocks in the function. Which is to say, if you have something like:
```
def foo():
    yield 10
    try:
        return
     except:
        print('hi!')
```
the except block shouldn't run.

This shouldn't be too hard to fix, I think. The exception just needs to be raised from a basic block that doesn't have an `error_handler` set.